inception/inception-api-annotation-experimental/src/main/java/de/tudarmstadt/ukp/inception/experimental/api/AnnotationSystemAPIImpl.java

```diff
@@ -17,402 +17,669 @@
+                    aUpdateArcRequest.getFeatures());
 
-        }  catch (Exception e) {
-            e.printStackTrace();
-            createErrorMessage(e.getMessage(), aClientMessage.getClientName());
+            this.annotationProcessAPI.sendUpdateArc(response,
+                    String.valueOf(aUpdateArcRequest.getProjectId()),
+                    String.valueOf(aUpdateArcRequest.getDocumentId()));
+
+        }
+        catch (Exception e) {
+            LOG.error("Exception has been thrown during handleUpdateRelation()", e);
+            throw new Exception();
         }
     }
 
     @Override
-    public CAS getCasForDocument(String aUser, long aProject, long aDocument)
+    public void handleCreateArc(CreateArcRequest aCreateArcRequest) throws IOException
     {
-        try {
-            CasStorageSession.open();
+        try (CasStorageSession session = CasStorageSession.open()) {
             MDC.put(Logging.KEY_REPOSITORY_PATH, repositoryProperties.getPath().toString());
-            SourceDocument sourceDocument = documentService.getSourceDocument(aProject, aDocument);
+            SourceDocument sourceDocument = documentService.getSourceDocument(
+                    aCreateArcRequest.getProjectId(), aCreateArcRequest.getDocumentId());
 
-            CAS cas = documentService.readAnnotationCas(sourceDocument, aUser);
+            CAS cas = documentService.readAnnotationCas(sourceDocument,
+                    aCreateArcRequest.getClientName());
 
-            CasStorageSession.get().close();
-            return cas;
+            AnnotationFS source = selectAnnotationByAddr(cas,
+                    aCreateArcRequest.getSourceId().getId());
+            AnnotationFS target = selectAnnotationByAddr(cas,
+                    aCreateArcRequest.getTargetId().getId());
+
+            TypeAdapter adapter = annotationService
+                    .getAdapter(
+                            annotationService
+                                    .getLayer(annotationService
+                                            .listAttachedRelationLayers(
+                                                    getLayer(aCreateArcRequest.getProjectId(),
+                                                            aCreateArcRequest.getLayer()))
+                                            .get(0).getId()));
+
+            ((RelationAdapter) adapter).add(
+                    documentService.getSourceDocument(aCreateArcRequest.getProjectId(),
+                            aCreateArcRequest.getDocumentId()),
+                    aCreateArcRequest.getClientName(), source, target, cas);
         }
         catch (Exception e) {
-            e.printStackTrace();
-            CasStorageSession.get().close();
-            return null;
+            LOG.error("Exception has been thrown during handleCreateRelation()", e);
+            createErrorMessage(e.getMessage(), aCreateArcRequest.getClientName());
```

Comment:
No, createErrorMessage() actually sends the message in its method-body. I only created a separate method here because of redundancy as nearly all classes contain that piece of code otherwise